src/Solvers/solve_poisson_equation.jl

```diff
@@ -0,0 +1,23 @@
+function solve_poisson_equation!(solver)
+    topo = TX, TY, TZ = topology(solver.grid)
+    λx, λy, λz = solver.eigenvalues
+
+    # We can use the same storage for the RHS and the solution ϕ.
+    RHS, ϕ = solver.storage, solver.storage
+
+    # Apply forward transforms in order
+    [transform!(RHS, solver.buffer) for transform! in solver.transforms.forward]
+
+    # Solve the discrete Poisson equation.
+    @. ϕ = -RHS / (λx + λy + λz)
+
+    # Setting DC component of the solution (the mean) to be zero. This is also
+    # necessary because the source term to the Poisson equation has zero mean
+    # and so the DC component comes out to be ∞.
+    CUDA.@allowscalar ϕ[1, 1, 1] = 0
```

Comment:
Isn't the problem that `λx[1, 1, 1] + λy[1, 1, 1] + λz[1, 1, 1] = 0`? If `RHS[1, 1, 1] = 0` we get `NaN`, otherwise we'd get `Inf` (and we can't have either). I'm not 100% sure what DC refers to but I think it's sufficient to mention that, in eigenspace, `ϕ[1, 1, 1]` is the "zeroth mode" corresponding to the volume mean of the transform of `ϕ`, or of `ϕ` in physical space.

Reply:
Yeah exactly, so you get one NaN which then turns the entire array into NaNs once you apply an inverse transform.

Ah sorry DC component = direct current component lol, I guess referring to the fact that there's a non-zero mean.

The physical reasoning I've had for why this step is needed is that solutions to Poisson's equation are only unique up to a constant (the global mean of the solution), so we need to pick a constant. `ϕ[1, 1, 1] = 0` chooses the constant to be zero so that the solution has zero-mean.

I guess we always take gradients of the pressure so the constant theoretically shouldn't matter.

I'll improve the comment and add your suggestion.

Reply:
Thanks for explaining as I thought DC was discrete cosine, but since we are talking about the mean, I guess that would be the first component of the discrete cosine transform?

Reply:
I think it's there whether we're doing FFTs for `Periodic` or DCTs for `Bounded` but yeah I think in both cases it's the first (zeroth?) component of the transform.

Reply:
> The physical reasoning I've had for why this step is needed is that solutions to Poisson's equation are only unique up to a constant

I agree with this! We also know that adding a constant to pressure leaves the problem unchanged (at least for the equation of state we use...)